src/lib/fields_derivers_snapps/fields_derivers_snapps.ml

```diff
@@ -211,6 +211,35 @@ module Make (Schema : Graphql_intf.Schema) = struct
 
   let arg_typ obj = !(obj#graphql_arg) ()
 
+  let rec json_to_safe : Yojson.Basic.t -> Yojson.Safe.t = function
+    | `Assoc kv ->
+        `Assoc (List.map kv ~f:(fun (k, v) -> (k, json_to_safe v)))
+    | `Bool b ->
+        `Bool b
+    | `Float f ->
+        `Float f
+    | `Int i ->
+        `Int i
+    | `List xs ->
+        `List (List.map xs ~f:json_to_safe)
+    | `Null ->
+        `Null
+    | `String s ->
+        `String s
+
+  (* Builds the graphql-expected output query shape
+   * (essentially the keys of all the nested JSON) *)
+  let rec json_keys : Yojson.Safe.t -> string = function
```

Comment:
Is this really correct? It seemed fine to do that in a test, but in a more general situation it seems fragile to get the query structure by dynamically pulling out keys of a dummy json object. Are all relevant sub-fields really not null, are all arrays really non-empty? And if not, won't our query miss some fields?

Reply:
Intuitively, my preference would be to have logic in graphql-derivers that can produce the json_keys

Reply:
True it is a bit brittle as is… I guess it’s time for a `to_graphql_query` — should be really easy at least